### ISSUE TITLE
Make it possible to run our tests concurrently

### ISSUE THREAD

**Issue by carlspring** [strongbox/strongbox#839]
We are now able to build modules of our multi-module projects in parallel with the Maven command line option `-T` .

We should try to parallelize the tests in our modules, in order to be able to run them concurrently per module (https://maven.apache.org/surefire/maven-surefire-plugin/examples/fork-options-and-parallel-execution.html), if possible.

**Comment by amogozov** [comment]
Hi @carlspring , I would like to work on this

**Comment by carlspring** [comment]
Hi!

Great! Please, feel free to experiment with it and report your finding and fixes!

Good luck! :)

**Comment by carlspring** [comment]
@amogozov : Any luck with this?

**Comment by amogozov** [comment]
Hey @carlspring . I did some investigation, I think that concurrent test execution is enabled by default when you pass the `-T` parameter. For example
> mvn clean -T 1C install 

(meaning that one JVM will be forked to run one test class, up to number of cores) finished on my MacBook Pro in 4:40. Whereas 
> mvn clean install 

took 5:11.

Would you like to run tests concurrently even when `-T` param is not provided?

**Comment by carlspring** [comment]
@amogozov / @ptirador , it would be interesting to see how things behave once #871 has been merged into the `master`.

@amogozov : Perhaps you might like to try it from the branch and report back?

**Comment by ptirador** [comment]
It seems that parallel test execution is availiable as an experimental feature of JUnit 5.3:
[Link](https://junit.org/junit5/docs/snapshot/user-guide/#writing-tests-parallel-execution)

**Comment by carlspring** [comment]
@amogozov ,

Would you like to join in and help out with the refactoring of the tests in these modules?
